terraform-environments/hosts-user-data.sh

```diff
@@ -2,13 +2,13 @@
+  encoded_python_script="IyBVc2UgdGhpcyBjb2RlIHNuaXBwZXQgaW4geW91ciBhcHAuCiMgSWYgeW91IG5lZWQgbW9yZSBpbmZvcm1hdGlvbiBhYm91dCBjb25maWd1cmF0aW9ucwojIG9yIGltcGxlbWVudGluZyB0aGUgc2FtcGxlIGNvZGUsIHZpc2l0IHRoZSBBV1MgZG9jczoKIyBodHRwczovL2F3cy5hbWF6b24uY29tL2RldmVsb3Blci9sYW5ndWFnZS9weXRob24vCgppbXBvcnQgYm90bzMKZnJvbSBib3RvY29yZS5leGNlcHRpb25zIGltcG9ydCBDbGllbnRFcnJvcgppbXBvcnQganNvbgoKCmRlZiBnZXRfc2VjcmV0KCk6CgogICAgc2VjcmV0X25hbWUgPSAicHJvZC9ob3N0cy9rZXlzIgogICAgcmVnaW9uX25hbWUgPSAidXMtZWFzdC0xIgoKICAgICMgQ3JlYXRlIGEgU2VjcmV0cyBNYW5hZ2VyIGNsaWVudAogICAgc2Vzc2lvbiA9IGJvdG8zLnNlc3Npb24uU2Vzc2lvbigpCiAgICBjbGllbnQgPSBzZXNzaW9uLmNsaWVudCgKICAgICAgICBzZXJ2aWNlX25hbWU9J3NlY3JldHNtYW5hZ2VyJywKICAgICAgICByZWdpb25fbmFtZT1yZWdpb25fbmFtZQogICAgKQoKICAgIHRyeToKICAgICAgICBnZXRfc2VjcmV0X3ZhbHVlX3Jlc3BvbnNlID0gY2xpZW50LmdldF9zZWNyZXRfdmFsdWUoCiAgICAgICAgICAgIFNlY3JldElkPXNlY3JldF9uYW1lCiAgICAgICAgKQogICAgZXhjZXB0IENsaWVudEVycm9yIGFzIGU6CiAgICAgICAgIyBGb3IgYSBsaXN0IG9mIGV4Y2VwdGlvbnMgdGhyb3duLCBzZWUKICAgICAgICAjIGh0dHBzOi8vZG9jcy5hd3MuYW1hem9uLmNvbS9zZWNyZXRzbWFuYWdlci9sYXRlc3QvYXBpcmVmZXJlbmNlL0FQSV9HZXRTZWNyZXRWYWx1ZS5odG1sCiAgICAgICAgcmFpc2UgZQoKICAgICMgRGVjcnlwdHMgc2VjcmV0IHVzaW5nIHRoZSBhc3NvY2lhdGVkIEtNUyBrZXkuCiAgICBzZWNyZXQgPSBnZXRfc2VjcmV0X3ZhbHVlX3Jlc3BvbnNlWydTZWNyZXRTdHJpbmcnXQoKICAgICMgY3JlYXRlIGEganNvbiBvYmplY3QgdXNpbmYgdGhlIHJldHVybmVkIHNlY3JldAogICAgZGF0YSA9IGpzb24ubG9hZHMoc2VjcmV0KQoKICAgICMgR2V0IHRoZSB2YWx1ZSBvZiBlYWNoIGtleQogICAgcHVibGljX2tleSA9IGRhdGFbJ3B1YmxpYyddCiAgICBwcml2YXRlX2tleSA9IGRhdGFbJ3ByaXZhdGUnXQoKICAgICMgWW91ciBjb2RlIGdvZXMgaGVyZS4KICAgIHByaW50KHB1YmxpY19rZXkpCiAgICBwcmludChwcml2YXRlX2tleSkKCmdldF9zZWNyZXQoKQoK"
```

Comment:
Same as before - I don't see the need to encode, if we can otherwise. If we *must* encode then sure, but otherwise it leaves holes open for a lot of errors